docs/spec.md

```diff
@@ -232,6 +236,42 @@ Informs the client about services that are no longer available. Only supported i
 }
 ```
 
+### Connection graph update
+
+Informs the client about updates to the connection graph. This is only sent to clients which have previously [subscribed to connection graph updates](#subscribe-connection-graph-updates). Only supported if the server previously declared that it has the `connectionGraph` [capability](#server-info).
+
+#### Fields
+
+- `op`: string `"connectionGraphUpdate"`
+- `topicUpdates`: array of newly added or updated topics
+  - `name`: string, topic name
+  - `publisherIds`: array of string, list of publisher IDs
+  - `subscriberIds`: array of string, list of subscriber IDs
+- `serviceUpdates`: array of newly added or updated services
+  - `name`: string, service name
+  - `providerIds`: array of string, list of service provider IDs
+- `removedTopics`: array of string, names of topics that have been removed
+- `removedServices`: array of string, names of services that have been removed
+
+#### Example
+
+```json
+{
+  "op": "connectionGraphUpdate",
+  "topicUpdates": [
+    {
+      "name": "/foo",
+      "publisherIds": ["/node_1"],
+      "subscriberIds": ["/node_2"]
+    },
+    { "name": "/bar", "publisherIds": ["/node_1"], "subscriberIds": [] }
+  ],
+  "serviceUpdates": [{ "name": "/set_bool", "providerIds": ["/node_2"] }],
+  "removedTopics": ["/baz"],
```

Comment:
This could also be communicated by a topic with zero publishers and zero subscribers

Reply:
It may be valuable to communicate the presence of a topic or service that has no publishers or subscribers (yet). I'm not sure if this is possible in ROS, but you could imagine a robot framework that uses a static configuration of topics before any nodes register as publishers or subscribers.